src/extension.ts

```diff
@@ -674,12 +674,15 @@ export class ExtensionManager implements vscode.Disposable {
                 }
                 // Inform cpptools that custom CppConfigurationProvider will be able to service the current workspace.
                 this.ensureCppToolsProviderRegistered();
-                if (cpptools.notifyReady && this.cpptoolsNumFoldersReady < this.projectController.numOfWorkspaceFolders) {
-                    ++this.cpptoolsNumFoldersReady;
-                    if (this.cpptoolsNumFoldersReady === this.projectController.numOfWorkspaceFolders) {
-                        // Notify cpptools that the provider is ready to provide IntelliSense configurations.
-                        cpptools.notifyReady(this.configProvider);
-                        this.configProvider.markAsReady();
+                if (cpptools.notifyReady) {
```

Comment:
This was a long standing bug. These conditions need to be separated.  Also only affected multi-root.

Reply:
The `else` case is for old versions of the config provider API (< 2.1) and we shouldn't have been going there for multi-root projects.